src/modules/interfaces.ts

```diff
@@ -397,6 +401,162 @@ export class RpcBase {
   formatObjKeysToSnakeCase(obj: any) {
     return Object.fromEntries(Object.entries(obj).map(([k, v]) => [snakeCase(k), v]));
   }
+
+  getInstallUninstallSql(key: string) {
+    const rpcOutputEntries = Object.entries(this.outputTable ?? {});
+    const rpcOutputTable = rpcOutputEntries
+      .map(([columnName, columnType]) => `${columnName} ${columnType}`)
+      .join(', ');
+    const afterInstallSql = `
+        create or replace function ${snakeCase(
+          key,
+        )}(variadic _args text[] default array[]::text[]) returns table (
+          ${rpcOutputTable}
+        )
+        language plpgsql security definer
+        as $$
+        declare
+          _content text;
+        begin
+          perform http.http_set_curlopt('CURLOPT_TIMEOUT_MS', '3600000');
+          select content into _content
+          from http.http_post(
+            'http://${config.http.host}:8088/v1/db/rpc',
+            json_build_object(
+              'dbId', current_database(),
+              'dbUser', SESSION_USER,
+              'params', _args,
+              'modulename', '${this.module.name}',
+              'methodname', '${key}',
+              'preTransaction', '${this.preTransactionCheck}',
+              'postTransaction', '${this.postTransactionCheck}'
+            )::varchar,
+            'application/json'
+          );
+          if json_typeof(_content::json) <> 'array' then
+            if json_typeof(_content::json) = 'object' then
+              raise exception 'Error: %', _content::json->>'message';
+            else
+              raise exception 'Error: %', _content;
+            end if;
+          end if;
+          return query select
+            ${
+              rpcOutputEntries.length
+                ? `${rpcOutputEntries
+                    .map(([col, typ]) => `try_cast(j.s->>'${col}', NULL::${typ}) as ${col}`)
+                    .join(', ')}`
+                : ''
+            }
+          from (
+            select json_array_elements(_content::json) as s
+          ) as j;
+        end;
+        $$;
+      `;
+    const beforeUninstallSql = `
+        DROP FUNCTION "${snakeCase(key)}";
+      `;
+    return { beforeUninstallSql, afterInstallSql };
+  }
+}
+
+export class AwsSdkInvoker extends RpcBase {
+  preTransactionCheck = PreTransactionCheck.NO_CHECK;
+  postTransactionCheck = PostTransactionCheck.NO_CHECK;
+
```

Comment:
As commented [here](https://github.com/iasql/iasql-engine/pull/1849#discussion_r1086934117) I think this should be:
```suggestion
  preTransactionCheck = PreTransactionCheck.WAIT_FOR_LOCK;
  postTransactionCheck = PostTransactionCheck.UNLOCK_ALWAYS;

```

Reply:
You're right. I changed it to the way you suggested.